src/sql/src/func.rs

```diff
@@ -1634,6 +1640,31 @@ macro_rules! builtins {
     }};
 }
 
+/// Constructs builtin function map with potentially dangerous functions.
+macro_rules! builtins_dangerous {
+    {
+        $(
+            $name:expr => $ty:ident {
+                $($params:expr => $op:expr => $return_type:expr, $oid:expr, $dangerous:expr;)+
+            }
+        ),+
+    } => {{
+
+        let mut builtins = BTreeMap::new();
+        $(
+            let impls = vec![$(impl_def!($params, $op, $return_type, $oid, $dangerous)),+];
+            let func = Func::$ty(impls);
+            let expect_set_return = matches!(&func, Func::Table(_));
+            for imp in func.func_impls() {
+                assert_eq!(imp.return_is_set, expect_set_return, "wrong set return value for func with oid {}", imp.oid);
+            }
+            let old = builtins.insert($name, func);
+            assert!(old.is_none(), "duplicate entry in builtins list {:?}", old);
+        )+
+        builtins
+    }};
+}
+
```

Comment:
Is there any way to combine this with the macro above?

Reply:
Essentially I want to overload he inner repeated pattern.

Reply:
I don't know enough about macros to say. Might need to ask a rust person.